### PR TITLE
Made The Code Clearer

### PR DESCRIPTION
**Pull request by ranbir7**
Codes and lines were made smaller to make it _readable_ and some type hints are added.

### CODE REVIEW

File: autoprof/image/image_object.py [hunk 1]
```diff
@@ -107,40 +136,60 @@ def shape(self):
...
+        """
         return self._data
     @data.setter
-    def data(self, data):
+    def data(self, data)-> None:
+        """Set the image data."""
         self.set_data(data)
         
-    def set_data(self, data, require_shape = True):
+    def set_data(self, data: Union[torch.Tensor, np.ndarray], require_shape: bool = True):
```
Comment (ranbir7) [inline autoprof/image/image_object.py:172]: is the bool here making a problem?

**Comment by ranbir7** [comment]
Do check and tell me whats happening? The tests say the recursion depth is exceeded..

File: autoprof/image/image_object.py [hunk 2]
```diff
@@ -67,29 +65,60 @@ def __init__(self,
...
+        
         assert not (pixelscale is None and window is None)
+
+        #set the data
         self.data = data
-        self.zeropoint = None if zeropoint is None else torch.as_tensor(zeropoint, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
+
+        #set Zeropoint
+        if zeropoint is None:
+            self.zeropoint = none
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: should be `self.zeropoint = None`

File: autoprof/image/image_object.py [hunk 3]
```diff
@@ -67,29 +65,60 @@ def __init__(self,
...
         else:
+            #When The Window object is provided
             self.window = window
             if pixelscale is None:
                 self.pixelscale = self.window.shape[0] / self.data.shape[1]
             else:
-                self.pixelscale = torch.as_tensor(pixelscale, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
+                self.pixelscale = torch.as_tensor(pixelscale, 
+                                                 dtype = AP_config.ap_dtype, 
+                                                 device = AP_config.ap_device)
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: I think an extra space is needed here for the indentation

File: autoprof/image/image_object.py [hunk 4]
```diff
@@ -107,40 +136,60 @@ def shape(self):
...
-        """Determine if the center of the image is aligned at a pixel center
-        (True) or if it is aligned at a pixel edge (False).
+    def center_alignment(self)-> torch.Tensor:
+        """Determine if the center of the image is aligned at a pixel center (True) 
+           or if it is aligned at a pixel edge (False).
 
         """
-        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.tensor(0.5, dtype = AP_config.ap_dtype, device = AP_config.ap_device), atol = 0.25)
+        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, 
+                            0.5,
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: Looks like this line got mixed up. See the previous version, the `0.5` was wrapped in `torch.tensor` along wiht some other adjustments

File: autoprof/image/image_object.py (hunk 1 above)
Comment (ConnorStoneAstro) [reply autoprof/image/image_object.py:172]: The `bool` looks ok to me

File: autoprof/image/image_object.py [hunk 5]
```diff
@@ -107,40 +136,60 @@ def shape(self):
...
+            
+        Raises:
+            AssertionError: If `require_shape` is `True` and the shape of the data is different from the current data.
+        """
         if self._data is not None and require_shape:
             assert data.shape == self._data.shape
-        self._data = data.to(dtype = AP_config.ap_dtype, device = AP_config.ap_device) if isinstance(data, torch.Tensor) else torch.as_tensor(data, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
-        
+        if isinstance(data, torch.Tensor):
+            self.data = data.to(dtype = AP_config.ap_dtype, device=AP_config.ap_device)
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: This is where the recursion is happening. Be careful to use `self._data` not `self.data` when setting the value

File: autoprof/image/image_object.py [hunk 6]
```diff
@@ -107,40 +136,60 @@ def shape(self):
...
+        else:
+            self.data = torch.as_tensor(data, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: Same here, should be `self._data` at the beginning

**Review by ConnorStoneAstro — COMMENTED** [review]
These are some great updates! I found where the recursion error is happening. Try fixing these points and lets see if it works!

**Comment by ranbir7** [comment]
tell me whats happening though?

File: autoprof/image/image_object.py [hunk 7]
```diff
@@ -107,40 +136,56 @@ def shape(self):
...
-    def center_alignment(self):
-        """Determine if the center of the image is aligned at a pixel center
-        (True) or if it is aligned at a pixel edge (False).
+    def center_alignment(self)-> torch.Tensor:
+        """Determine if the center of the image is aligned at a pixel center (True) 
+           or if it is aligned at a pixel edge (False).
 
         """
-        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.tensor(0.5, dtype = AP_config.ap_dtype, device = AP_config.ap_device), atol = 0.25)
+        return torch.isclose(((self.center - self.origin) / self.pixelscale) % 1, torch.Tensor(0.5, atol = 0.25, dtype = AP_config.ap_dtype, device = AP_config.ap_device))
```
Comment (ConnorStoneAstro) [inline autoprof/image/image_object.py]: Looks like the `atol = 0.25` argument has been put inside the tensor creation. It should come after the tensor is created

**Comment by ranbir7** [comment]
the torch.tensor was creating problems 
I kept it as torch. _T_ ensor 
whereas I should have kept the _T_ small..
lets see if it works or not

**Comment by ranbir7** [comment]
Perfect !
Merge it , I'll be adding more comments and docs later
Thanks for the help !!

**Review by ConnorStoneAstro — APPROVED** [review]
Everything looks great!

**Pull request merged**